pkg/kubectl/cmd/annotate.go

```diff
@@ -114,6 +114,7 @@ func NewCmdAnnotate(f *cmdutil.Factory, out io.Writer) *cobra.Command {
 	cmd.Flags().StringVar(&options.resourceVersion, "resource-version", "", "If non-empty, the annotation update will only succeed if this is the current resource-version for the object. Only valid when specifying a single resource.")
 	usage := "Filename, directory, or URL to a file identifying the resource to update the annotation"
 	kubectl.AddJsonFilenameFlag(cmd, &options.filenames, usage)
+	cmdutil.AddRecursiveFlags(cmd)
```

Comment:
Mismatch between how the flag is registered (encapsulated in a helper that assumes a global name) and used (literal string scattered in every command) is not a pattern we want to extend.